src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java

```diff
@@ -80,6 +81,7 @@ public class ClientDevicesAuthService extends PluginService {
     private final GroupManager groupManager;
 
     private final CertificateManager certificateManager;
+    private final RegistryRefreshMonitor registryRefreshMonitor;
```

Comment:
There's a lot of functionality in this PR. Can you split it up? Here's what I'm envisioning:

1. refactor: add ThingRegistry -- just get the basic infrastructure in place
2. feat: add caching to enable offline auth -- continue to make cloud calls, but remember previous results if the cloud is unavailable
3. feat: add caching TTL -- cache eviction and configuration
4. feat: add cache warmer
5. feat: disk backed caching